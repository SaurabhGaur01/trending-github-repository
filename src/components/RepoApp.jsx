import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import Content from './Content/Content';
import handleRetrieveRepoData from '../thunks/handleRetrieveRepoData';

class RepoApp extends React.Component{
    componentDidMount(){
        const { actionCallRepoApi } = this.props;
        actionCallRepoApi();
    }

    render(){
        return (
            <React.Fragment>
                <Content />
            </React.Fragment>    
        )
    }
}

RepoApp.propTypes = {
    actionCallRepoApi: PropTypes.func.isRequired,
};

const mapDispatchAsProps = dispatch => {
    return {
        actionCallRepoApi: () => dispatch(handleRetrieveRepoData()),
    };
}

const hocChain = compose(
    connect(null, mapDispatchAsProps),
);

export { RepoApp as TestableRepoApp };

export default hocChain(RepoApp);