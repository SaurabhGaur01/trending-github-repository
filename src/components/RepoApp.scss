.main-card {
    display: flex;
    flex-wrap: wrap;
    margin-top: 12px;
    background-color: lightslategray;
}

.card-container {
    margin: 3px 10px 8px;
    height: 350px;
    width: 300px;
}

.heading-label {
    font-weight: 700;
    padding-top: 10px;
}

.description{
    padding-top: 10px;
}

.first-section {
    height: 172px;
    position: relative;
}

.star-count {
    color: rgba(0,0,0,.4);
}

.url {
    margin-left: 80px;
}

.language-selector {
    display: flex;
    margin-left: auto;
    background-color: white;
}

.container-head {
    display: flex;
}

.cus-fields {
    min-width: 120px !important;
}