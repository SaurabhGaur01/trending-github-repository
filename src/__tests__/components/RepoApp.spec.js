import React from 'react';
import { shallow } from 'enzyme';
import { TestableRepoApp } from '../../components/RepoApp';

describe('RepoApp Component', () => {
    let props;

    beforeEach(() =>{
        props = {
            actionCallRepoApi: jest.fn(),
        }
    });

    it('should render correctly', () => {
        const renderedModule = shallow(<TestableRepoApp {...props} />);
        expect(renderedModule).toMatchSnapshot();
    });

    describe('componentDidMount', () => {
        it('should call actionCallRepoApi', () => {
            const instance = shallow(<TestableRepoApp {...props} />).instance();
            instance.componentDidMount();
            expect(props.actionCallRepoApi).toHaveBeenCalled();
        })
    })
})