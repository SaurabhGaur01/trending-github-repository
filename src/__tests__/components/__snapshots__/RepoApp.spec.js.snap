// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoApp Component should render correctly 1`] = `
<Fragment>
  <Content />
</Fragment>
`;
